client/lib/wp/offline-library/index.js

```diff
@@ -0,0 +1,102 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import stringify from 'json-stable-stringify';
+
+const readCache = () => {
+	try {
+		return require( '../../../../cached-requests.json' );
+	} catch ( e ) {
+		return {};
+	}
+};
+
+const saveRequests = requests => {
+	const requestData = {};
+
+	requests.forEach( ( value, key ) => {
+		const path = JSON.parse( key ).path;
+
+		if ( ! requestData.hasOwnProperty( path ) ) {
+			requestData[ path ] = {};
+		}
+
+		requestData[ path ][ key ] = value;
+	} );
+
+	const file = new Blob( [ JSON.stringify( requestData, null, 2 ) ], {
+		type: 'application/json',
+	} );
+
+	const url = URL.createObjectURL( file );
+	window.open( url );
+};
+
+export const makeOffline = wpcom => {
+	const location = window.location;
+	const primingRequested = /[&?]wpcom_priming=1/.test( location );
+	const offlineRequested = /[&?]wpcom_offline=1/.test( location );
+
+	if ( ! primingRequested && ! offlineRequested ) {
+		return wpcom;
+	}
+
+	// eslint-disable-next-line no-console
+	primingRequested && console.log( 'Priming wpcom request cache' );
```

Comment:
should also skip printing this if offline is requested since you can only do one or the other

Reply:
skipped in 9107268